Allow inline message for but describe

mtsgrd opened the pull request:
Add an optional -m/--message flag to the describe command so users can
supply a new commit message or branch name on the command line instead
of opening an editor. This change threads the optional message through
the describe handlers and uses it to validate and apply the supplied
value, when absent it falls back to the existing editor-based flow.

Add tests for 'but describe' commit message handling.

mtsgrd committed:
Allow inline message for but describe

Add an optional -m/--message flag to the describe command so users can
supply a new commit message or branch name on the command line instead
of opening an editor. This change threads the optional message through
the describe handlers and uses it to validate and apply the supplied
value, when absent it falls back to the existing editor-based flow.

Byron commented on crates/but/tests/but/command/describe.rs:
```diff
@@ -0,0 +1,154 @@
+
+// Note: Branch rename test is omitted because the test scenario uses single-character
+// branch names ("A") which don't meet the 2-character minimum requirement for CLI IDs.
+// The branch rename functionality with -m flag is tested manually and works correctly.
```
@jonathantanmy2 

I thought this would be interesting to you - is it CLI IDs that have to adjust, or is it the commands that should be able to fallback to "could this be a very short name for an entity"? To me it feels that CLI Ids could handle this, as they usually have to know the entire set of possible entities and thus can deal with short names, which it seems like could just be an ID on their own.

Byron reviewed (commented):
Thanks for the invite - I will push some minor refactors here and merge the result.

Also, I realise that the commit-details function used here is only legacy because it uses legacy types, and it would take a final push to be able to turn it into non-legacy with the api-cmd macro so that `ProjectId` style functions can be become 'modern'. So I think I'd want to do that this week as well.

Byron committed:
refactor

- deduplicate
- remove unhappy tests in favor of unit-testing the respective function.

Byron committed:
Try to guide AI more so it writes better tests

Byron commented:
The main behavioural change I made was that changing something to the same message is not an error. If the user says they want it to be message "X", and it is message "X", it's what they want which is good - that's the motivating factor behind that.
One could also say that no-change is a good change. Similar behaviour is already implemented in "branch apply" - applying the same branch again does nothing, and is happy about that.

Besides that, generally we'd want to limit these slow (but fortunately not flaky) tests to the bare minimum, typically happy paths *and* UX or messages that are really important to us. What happens when it can't find something might be something we want to test, but I have a feeling that with some framework around selecting stacks, commits, and hunks we can test this centrally and don't have to test it for each command.

Lastly, I think AI does a fine job but should do better when it comes to deduplication, and of course not spamming tests.
